nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/test/java/org/apache/nifi/processors/kudu/TestScanKudu.java

```diff
@@ -0,0 +1,370 @@
+    public static final String DEFAULT_MASTERS = "testLocalHost:7051";
+
+    @Rule
+    public KuduTestHarness harness = new KuduTestHarness(
+        new MiniKuduCluster.MiniKuduClusterBuilder()
+            .addMasterServerFlag("--use_hybrid_clock=false")
+            .addTabletServerFlag("--use_hybrid_clock=false")
+    );
+
+    @Before
+    public void setup() throws InitializationException {
+        kuduScan = new MockScanKudu();
+        kuduScan.kuduClient = harness.getClient();
+        runner = TestRunners.newTestRunner(kuduScan);
+
+        setUpTestRunner(runner);
+    }
+
+    private void setUpTestRunner(TestRunner testRunner) throws InitializationException {
+        testRunner.setProperty(PutKudu.KUDU_MASTERS, DEFAULT_MASTERS);
+        testRunner.setProperty(ScanKudu.TABLE_NAME, DEFAULT_TABLE_NAME);
+    }
+
+    @Test
+    public void testKuduProjectedColumnsValidation() {
+        runner.setProperty(ScanKudu.TABLE_NAME, DEFAULT_TABLE_NAME);
+        runner.setProperty(ScanKudu.PREDICATES, "column1=val1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "c1,c2");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "c1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "c1 c2,c3");
+        runner.assertNotValid();
+
+        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "c1:,c2,c3");
+        runner.assertNotValid();
+
+        runner.setProperty(ScanKudu.PROJECTED_COLUMNS, "c1,c1,");
+        runner.assertNotValid();
+    }
+
+    @Test
+    public void testKuduPredicatesValidation() {
+        runner.setProperty(ScanKudu.TABLE_NAME, DEFAULT_TABLE_NAME);
+        runner.setProperty(ScanKudu.PREDICATES, "column1=val1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PREDICATES, "column1>val1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PREDICATES, "column1<val1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PREDICATES, "column1<=val1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PREDICATES, "column1>=val1");
+        runner.assertValid();
+
+        runner.setProperty(ScanKudu.PREDICATES, "column1=>=val1");
+        runner.assertNotValid();
+
+        runner.setProperty(ScanKudu.PREDICATES, "column1:val1");
+        runner.assertNotValid();
+    }
+
+    @Test
+    public void testNoIncomingFlowFile() {
+        runner.setProperty(ScanKudu.TABLE_NAME, DEFAULT_TABLE_NAME);
+        runner.setProperty(ScanKudu.PREDICATES, "column1=val1");
+
+        runner.run(1, false);
+        runner.assertTransferCount(ScanKudu.REL_FAILURE, 0);
+        runner.assertTransferCount(ScanKudu.REL_SUCCESS, 0);
+        runner.assertTransferCount(ScanKudu.REL_ORIGINAL, 0);
+    }
+
+    @Test
+    public void testInvalidKuduTableName() {
+        runner.setProperty(ScanKudu.TABLE_NAME, "${table1}");
```

Comment:
This is still basically the same as `testInvalidPredicatesPattern`. Both fail because the table name is invalid (resolved to empty), and actually both have invalid predicates as well (as both reference a non-existing column in a non-existing table).
Both tests are missing the table initialization.